### PR TITLE
removing root requirement

### LINKED ISSUE #7: using root user for everything feels like it could cause issues
I wanted to figure out if there is any interest in making a change to sparrow-patterns to not use the root user by default.  I have been playing around with this in my fork and I think I have it mostly working although there is still one issue where I have to specify the user_id in two places, but before getting into anything too technical, is this something you would want to change or is using the root user in docker containers fine for all of your use-cases.

### PR DESCRIPTION
**Pull request by kevinbird15**
Closes #7 

This is a starting point for the non-root version.  This could probably use more testing, but I wanted to at least put it out as a starting point and for somebody else to build on if anybody is interested.  

* I am not able to dynamically specify a username in devcontainer.json. This may not end up mattering since you can inject the username in the devcontainer template which should give you plenty of flexibility.  

I didn't modify any of the templates or code coming out of the sparrow-patterns lib directly at this point.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,29 +1,37 @@
 FROM python:3.9
 
-RUN SNIPPET="export PROMPT_COMMAND='history -a' && export HISTFILE=/commandhistory/.bash_history" && echo $SNIPPET >> "/root/.bashrc"
+ARG USERNAME=kevin
```
Comment (jbencook) [inline Dockerfile:3]: I don't mind hardcoding the username at all (that's what we currently have with `root` in effect). Could we change it to something generic like `dev` or similar?

**Comment by jbencook** [comment]
Thanks Kevin! I'll test this out as soon as I can.

File: Dockerfile (hunk 1 above)
Comment (kevinbird15) [reply Dockerfile]: Made this modification.  Good idea on dev.  I like that

File: Dockerfile [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
 
 ENV LANG=C.UTF-8 \
   LC_ALL=C.UTF-8 \
   PATH="${PATH}:/root/.poetry/bin"
 
+RUN apt update -y && apt install -y sudo
+RUN groupadd --gid $USER_GID $USERNAME &&\
+ useradd --uid $USER_UID --gid $USER_GID -m $USERNAME &&\  
+ echo ${USERNAME} ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/${USERNAME} &&\
+ chmod 0440 /etc/sudoers.d/${USERNAME}
```
Comment (jbencook) [inline Dockerfile:15]:
```suggestion
 chmod 0440 /etc/sudoers.d/${USERNAME} && chsh ${USERNAME} -s /bin/bash
```

This makes `bash` the default shell for the `dev` user. That way keyboard shortcuts still work.

Comment (kevinbird15) [reply Dockerfile]: Fixed this change.  The only difference is I put chsh on its own line so it matches the rest of the command. This is definitely a nice addition.  Thanks for figuring out how to do it!

Comment (kevinbird15) [reply Dockerfile]: @jbencook do you have a preference on resolving conversations?  I think it is best to have the person that opened the suggestion resolve but wanted to double check your thoughts on that.

Comment (jbencook) [reply Dockerfile]: Hm yeah that's a good question -- I've usually done the reverse (the person who addresses the comment resolves it), but I definitely see what you mean and I don't have a strong preference either way

**Review by jbencook — APPROVED** [review]
This is a big improvement -- thanks @kevinbird15!

**Pull request merged**